probe-rs/src/coresight/memory/adi_v5_memory_interface.rs

```diff
@@ -166,19 +201,66 @@ impl ADIMemoryInterface {
         let tar = TAR { address };
         self.write_register_ap(debug_port, tar)?;
 
-        for data in data.iter_mut() {
-            *data = self.read_register_ap(debug_port, DRW::default())?.data;
+        // figure out how many words we can write before the
+        // data overflows
+
+        // maximum chunk size
+        let max_chunk_size_bytes = 0x400_usize;
+
+        let mut remaining_data_len = data.len();
+
+        let first_chunk_size_bytes = std::cmp::min(
+            max_chunk_size_bytes - (address as usize % max_chunk_size_bytes),
+            data.len() * 4,
+        );
+
+        let mut data_offset = 0;
+
+        debug!(
```

Comment:
We should use log::debug!() here or change all the occurences accross the code (which i personnaly dislike as I think log::debug!() is more readable).